### ISSUE TITLE
Problem with ordering in producer and high throughput

### ISSUE THREAD

**Issue by Arfey** [dpkp/kafka-python#2025]
hi 🤚It's just a question.

Now I can't use idempotent producer, so keep ordering i need to set `max_in_flight_requests_per_connection` to 1.

Let's say that i have 1 broker and 6 partitions for some topic. So i write a lot of data to my topic via `test_producer`.

Each circle of `run_once` will generate list of buckets

https://github.com/dpkp/kafka-python/blob/87fb1bb48f82bcaa6c5e1a1edadab2832659801c/kafka/producer/sender.py#L116

and after that try to send these 

https://github.com/dpkp/kafka-python/blob/87fb1bb48f82bcaa6c5e1a1edadab2832659801c/kafka/producer/sender.py#L148-L152

we'll have only one connection because we have only one broker (connection generate for node_id and we have only one node)

`send` method have a two important pleases

first it's test for limit of `in_flight_requests` for current connection

https://github.com/dpkp/kafka-python/blob/3aada777e9c3bbb5751a15b615d6fbe4693cc6f0/kafka/conn.py#L1040-L1043

and add generated feature to `in_flight_requests`

https://github.com/dpkp/kafka-python/blob/3aada777e9c3bbb5751a15b615d6fbe4693cc6f0/kafka/conn.py#L973

so, first bucket will send but all others will drop with error `TooManyInFlightRequests`.

Yes, `max_in_flight_requests_per_connection=1` solve that if i will have 1 leader partition per broker but if it is not true i will get `TooManyInFlightRequests` error.

Generated connection per partition (not per node) should fix that.

am i right? and if yes how can i fix that?

**Comment by dpkp** [comment]
No, all partitions/batches that are destined for the same broker will be grouped into a single produce request. So there should not be a `TooManyInFlightRequests`. Please paste a traceback if you are seeing this error.
